modules/virtualisation/podman/podman.nix

```
@@ -0,0 +1,276 @@
+      type = with types; listOf package;
+      default = [ ];
+      example = lib.literalExample ''
+        [
+          pkgs.gvisor
+        ]
+      '';
+      description = ''
+        Extra packages to be installed in the Podman wrapper.
+      '';
+    };
+
+    finalPackage = lib.mkOption {
+      type = types.package;
+      default = podmanPackage;
+      internal = true;
+      readOnly = true;
+      description = ''
+        The final Podman package (including extra packages).
+      '';
+    };
+
+    defaultNetwork.extraPlugins = lib.mkOption {
+      type = types.listOf json.type;
+      default = [ ];
+      description = ''
+        Extra CNI plugin configurations to add to podman's default network.
+      '';
+    };
+
+    machines = lib.mkOption {
+      type = types.attrsOf (types.submodule machineOpts);
+      description =
+        "Virtual machine descriptions when podman is run in on non-linux systems";
+      default = {
+        podman-machine-default = {
+          active = true;
+          cpus = 2;
+          disk_size = 100;
+          memory = 2048;
+        };
+      };
+      example = {
+        intel-x86 = {
+          cpus = 2;
+          disk_size = 200;
+          memory = 4096;
+          image_path = "fedora-coreos-38.20230806.3.0-qemu.x86_64.qcow2.xz";
+          qemu_binary = "$${pkgs.qemu}/bin/qemu-system-x86_64";
+        };
+      };
+      # One and only one machine may be active at any given time
+      apply = machines:
+        assert ((lib.lists.count (machine: machine.active)
+          (lib.attrsets.attrValues machines)) == 1);
+        machines;
+    };
+
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      home.packages = [ cfg.finalPackage ]
+        ++ lib.optional cfg.enableDockerAlias dockerAlias;
+
+      virtualisation.containers = {
+        enable = true; # Enable common /etc/containers configuration
+        #        containersConf.settings = lib.optionalAttrs cfg.enableNvidia {
+        #          engine = {
+        #            conmon_env_vars =
+        #              [ "PATH=${lib.makeBinPath [ pkgs.nvidia-podman ]}" ];
+        #            runtimes.nvidia =
+        #              [ "${pkgs.nvidia-podman}/bin/nvidia-container-runtime" ];
+        #          };
+        #        };
+      };
+    }
+
+    (mkIf pkgs.stdenv.hostPlatform.isLinux (mkMerge [
+      {
+        systemd.user = {
+
+          services.podman = {
+            Unit = {
+              Description = "Podman API Service";
+              Requires = "podman.socket";
+              After = "podman.socket";
+              Documentation = "man:podman-system-service(1)";
+              StartLimitIntervalSec = 0;
+            };
+            Service = {
+              Type = "exec";
+              KillMode = "process";
+              Environment = ''LOGGING=" --log-level=info"'';
+              ExecStart =
+                [ "" "${cfg.finalPackage}/bin/podman $LOGGING system service" ];
```

```suggestion
                [ "${cfg.finalPackage}/bin/podman" "$LOGGING" "system" "service" ];
```

See my original PR(https://github.com/nix-community/home-manager/pull/2630#discussion_r797006402)